dev/env/scripts/lib.sh

```diff
@@ -0,0 +1,191 @@
+    export ENABLE_FM_PORT_FORWARDING=${ENABLE_FM_PORT_FORWARDING:-$ENABLE_FM_PORT_FORWARDING_DEFAULT}
+    export AUTH_TYPE=${AUTH_TYPE:-$AUTH_TYPE_DEFAULT}
+    export FINAL_TEAR_DOWN=${FINAL_TEAR_DOWN:-$FINAL_TEAR_DOWN_DEFAULT}
+
+    export FLEET_MANAGER_IMAGE="${FLEET_MANAGER_IMAGE:-$FLEET_MANAGER_IMAGE_DEFAULT}"
+    # When transferring images without repository hostname to Minikube it gets prefixed with "docker.io" automatically.
+    if [[ "$FLEET_MANAGER_IMAGE" =~ ^fleet-manager-.*/fleet-manager:.* ]]; then
+        export FULL_FLEET_MANAGER_IMAGE="docker.io/${FLEET_MANAGER_IMAGE}"
+    else
+        export FULL_FLEET_MANAGER_IMAGE="${FLEET_MANAGER_IMAGE}"
+    fi
+
+    verify_environment
+
+    disable_debugging
+    enable_debugging_if_necessary
+}
+
+disable_debugging() {
+    if [[ "$DEBUG" != "trace" ]]; then
+        set +x
+    fi
+}
+
+enable_debugging_if_necessary() {
+    if [[ "$DEBUG" != "false" ]]; then
+        set -x
+    fi
+}
+
+wait_for_container_to_appear() {
+    local namespace="$1"
+    local pod_selector="$2"
+    local container_name="$3"
+    local status=$($KUBECTL -n "$ACSMS_NAMESPACE" get pod -l "${pod_selector}" -o jsonpath="{.items[0].status.initContainerStatuses[?(@.name == '${container_name}')]} {.items[0].status.containerStatuses[?(@.name == '${container_name}')]}")
+    local state=$(echo "${status}" | jq -r ".state | keys[]")
+    log "Waiting for container ${container_name} within pod ${pod_selector} in namespace ${namespace} to appear..."
+    for i in $(seq 10); do
+        if [[ "$state" == "terminated" || "$state" == "running" ]]; then
+            echo "Container ${container_name} is in state ${state}"
+            break
+        fi
+        sleep 1
+    done
+}
+
+wait_for_container_to_become_ready() {
+    local namespace="$1"
+    local pod_selector="$2"
+
+    log "Waiting for pod ${pod_selector} within namespace ${namespace} to become ready..."
+
+    for i in $(seq 10); do
+        if $KUBECTL -n "$ACSMS_NAMESPACE" wait --timeout=5s --for=condition=ready pod -l "$pod_selector" 2>/dev/null >&2; then
+            break
+        fi
+        sleep 1
+    done
+    $KUBECTL -n "$ACSMS_NAMESPACE" wait --timeout=20s --for=condition=ready pod -l "$pod_selector"
+    sleep 1
+    log "Pod ${pod_selector} is ready."
+}
+
+wait_for_resource_to_appear() {
+    local namespace="$1"
+    local kind="$2"
+    local name="$3"
+
+    log "Waiting for ${kind}/${name} to be created in namespace ${namespace}"
+
+    for i in $(seq 10); do
+        if $KUBECTL -n "$namespace" get "$kind" "$name" 2>/dev/null >&2; then
+            return 0
+        fi
+        sleep 1
+    done
+
+    log "Giving up waiting for ${kind}/${name} in namespace ${namespace}"
+
+    return 1
+}
+
+wait_for_default_service_account() {
+    local namespace="$1"
+    if wait_for_resource_to_appear "$namespace" "serviceaccount" "default"; then
+        return 0
+    else
+        return 1
+    fi
```

Comment:
I think that should also work instead of the if (not tested)
```suggestion
    wait_for_resource_to_appear "$namespace" "serviceaccount" "default"
    return $?
```

Reply:
The "if" is for protecting the script from terminating due to the use of "set -e".
I agree this sounds weird, but the command within the if-condition is protected from early termination here and I don'g know of a better way.